ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/JsonRpcRequest.java

```diff
@@ -69,6 +70,15 @@ public String getVersion() {
     return params;
   }
 
+  @JsonIgnore
+  public void assertMaxLength(final int expectedLength) {
```

Comment:
I'd suggest reworking the structure of this code a bit by:
- Adding a method `int getMaxAllowedParams()` to the `JsonRpcMethod` interface
- Adding a `default` method `assertValidRequest(JsonRpcRequest request)` to `JsonRpcMethod` that throws an `InvalidJsonRcpParameters` exception as you have here if there are too many params
- Validating the request by calling `jsonRpcMethod.assertValidRequest(request)` before executing [`jsonRpcMethod.response(request)` in `JsonRpcHttpService`](https://github.com/PegaSysEng/pantheon/blob/fdfa35443bfe26a01919c358b74f1184d41c9b66/ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/JsonRpcHttpService.java#L497)

This will make it straightforward to consistently enforce this validation without expecting every method to run these checks manually.  And it allows methods to override the default validation if necessary.

Reply:
Thank you so much! Definitely a better code structure, I will do my best to implement it.
Quick question : should I still manually add a tooManyParams test for every method, or should I just add one test for one method and "assume" that it will be correct for other methods ?

Reply:
Seems kind of painful to handcraft tests for every method.  But if there was some way to "automatically" test each method, that would be cool.  One idea:
- Write a parameterized test ([example](https://github.com/PegaSysEng/pantheon/blob/fdfa35443bfe26a01919c358b74f1184d41c9b66/consensus/clique/src/test/java/tech/pegasys/pantheon/consensus/clique/headervalidationrules/VoteValidationRuleTest.java#L30-L42)) that runs for every method and constructs a request with 1 too many parameters, then verifies that the expected exception is thrown

Otherwise, I think we can probably get away with just adding a single test to `JsonRpcHttpServiceTest`.